File: Examples/VisualBasic/Articles/SetBackgroundPicture.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Cells
Imports System

Namespace Articles
    Public Class SetBackgroundPicture
        Public Shared Sub Run()
            ' ExStart:1
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Instantiate a new Workbook.
            Dim workbook As New Workbook()

            ' Get the first worksheet.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Define a string variable to store the image path.
            Dim ImageUrl As String = dataDir & "aspose-logo.jpg"

            ' Get the picture into the streams.
            Dim fs As FileStream = File.OpenRead(ImageUrl)

            ' Define a byte array.
            Dim imageData(fs.Length - 1) As Byte

            ' Obtain the picture into the array of bytes from streams.
            fs.Read(imageData, 0, imageData.Length)

            ' Close the stream.
            fs.Close()

            ' Set the background image for the sheet.
            sheet.SetBackground(imageData)

            ' Save the Excel file
            workbook.Save(dataDir & "BackImageSheet.out.xlsx")

            ' Save the HTML file
            workbook.Save(dataDir & "output.html", SaveFormat.Html)
            ' ExEnd:1
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Articles/SetBackgroundPicture.cs
using System.IO;

using Aspose.Cells;
using System;

namespace Aspose.Cells.Examples.CSharp.Articles
{
    public class SetBackgroundPicture
    {
        public static void Run()
        {
            // ExStart:1
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            //Output directory
            string outputDir = RunExamples.Get_OutputDirectory();

            // Instantiate a new Workbook.
            Workbook workbook = new Workbook();

            // Get the first worksheet.
            Worksheet sheet = workbook.Worksheets[0];

            // Set the background image for the sheet.
            sheet.BackgroundImage = File.ReadAllBytes(sourceDir + "background.jpg");

            // Save the Excel file
            workbook.Save(outputDir + "outputBackImageSheet.xlsx");

            // Save the HTML file
            workbook.Save(outputDir + "outputBackImageSheet.html", SaveFormat.Html);
            // ExEnd:1

            Console.WriteLine("SetBackgroundPicture executed successfully.");
        }
    }
}